### ISSUE TITLE
misleading message during prisma migrate dev --create-only

### ISSUE THREAD

**Issue by sairajchouhan** [prisma/prisma#7306]
### Bug description

when I change the column name and run `npx prisma migrate dev --create-only` the message shown in the console is that the data will be  deleted, but its job is to only create a migration file

![2021-05-26 23_11_01-schema prisma - postgres-prisma-todo - Visual Studio Code](https://user-images.githubusercontent.com/57995897/119834274-6edf0b00-bf1d-11eb-90ae-9c7f48270a84.png)

### Environment & setup

- OS: 
- Database: 
- Node.js version: 
- OS: Windows 10
- Database: PostgreSQL
- Node.js version: 14.16.1

### Prisma Version
```
2.23.0
```

**Comment by pantharshit00** [comment]
@sairaj2119 

Can you please provide a reproduction here? Provide the schema and the changes you made to the schema which lead to the above behavior. That will help us in debugging this issue.

**Comment by sairaj2119** [comment]
I tried to recreate the the process but in the mean while it worked so, I discussed this whole issue in slack with janpio and ryan and they said just to raise the issue no problem even if it is working

**Comment by ryands17** [comment]
@sairaj2119 👋 

The message is indeed misleading as `prisa migrate dev --create-only` doesn't delete data from the database.


@pantharshit00 To reproduce:

1. Create a sample schema:

```prisma
model User {
  id Int @id @default(autoincrement())
  name String
}
```

2. Run migrate: `prisma migrate dev`

3. Seed some sample data

4. Update the model

```prisma
model User {
  id Int @id @default(autoincrement())
  fullName String
}
```

4. Run `prisma migrate dev --create-only`. You will get a warning in the terminal that data will be lost. Press `y` to confirm.

5. Check the database and data still persists

The main issue here is we can change the messaging for `--create-only` as data is not lost when just creating the migration.

**Comment by pantharshit00** [comment]
Thanks for the excellent reproduction Ryan. I can confirm this.

**Comment by janpio** [comment]
Clearer example what this looks like in the console:

```
C:\Users\Jan\Documents\throwaway\planetscale>npx prisma migrate dev --create-only
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Datasource "db": MySQL database "fk-test" at "127.0.0.1:3306"

√ Drift detected: Your database schema is not in sync with your migration history.

We need to reset the MySQL database "fk-test" at "127.0.0.1:3306".
Do you want to continue? All data will be lost. ... no

Reset cancelled.
```

**Comment by janpio** [comment]
I fear there are also cases where it would indeed do some DDL statements:

```
C:\Users\Jan\Documents\throwaway\planetscale>npx prisma migrate dev --create-only
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Datasource "db": MySQL database "fk-test" at "127.0.0.1:3306"

√ Drift detected: Your database schema is not in sync with your migration history.

We need to reset the MySQL database "fk-test" at "127.0.0.1:3306".
Do you want to continue? All data will be lost. ... yes

Error: P3016

The fallback method for database resets failed, meaning Migrate could not clean up the database entirely. Original error: 
Database error
Error querying the database: Server error: `ERROR HY000 (1105): direct DDL is disabled'
   0: sql_migration_connector::best_effort_reset
           with connection=Connection(Mysql(MysqlUrl { url: Url { scheme: "mysql", username: "root", password: None, host: Some(Domain("127.0.0.1")), port: Some(3306), path: "/fk-test", query: None, fragment: None }, query_params: MysqlUrlQueryParams { ssl_opts: SslOpts { pkcs12_path: None, password: None, root_cert_path: None, skip_domain_validation: false, accept_invalid_certs: true }, connection_limit: None, use_ssl: false, socket: None, socket_timeout: None, connect_timeout: Some(5s), pool_timeout: Some(10s), max_connection_lifetime: None, max_idle_connection_lifetime: Some(300s) } }))
             at migration-engine\connectors\sql-migration-connector\src\lib.rs:99
   1: migration_core::api::Reset
             at migration-engine\core\src\api.rs:157
```

Fortunately my database does not allow that. I have no idea which DDL commands this would execute.

**Comment by tomhoule** [comment]
We should change the message, but the warnings are still applicable because it's what is going to happen when you later apply the migration that was created.

**Comment by thebiglabasky** [comment]
Due to capacity in the team we'll have to take another look at it in an upcoming sprint.

**Comment by janpio** [comment]
Seems in 3.0.2 the error message slightly changed - "Some data will be lost" is gone:

```
C:\Users\Jan\Documents\throwaway\73906>npx prisma migrate dev --create-only
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Datasource "db": SQLite database "dev.db" at "file:./dev.db"



⚠️ We found changes that cannot be executed:

  • Step 0 Added the required column `fullName` to the `User` table without a default value. There are 1 rows in this table, it is not possible to execute this step.


⚠️  Warnings for the current datasource:

  • You are about to drop the column `name` on the `User` table, which still contains 1 non-null values.

√ Are you sure you want create this migration? ... yes
```

1. As `--create-only` is used, obviously nothing was dropped in that moment.
2. Running `npx prisma migrate deploy` then indeed failed because of the missing default value

So running this again with a default value, leads to  the cleaner output:

```
C:\Users\Jan\Documents\throwaway\73906>npx prisma migrate dev --create-only
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Datasource "db": SQLite database "dev.db" at "file:./dev.db"


⚠️  Warnings for the current datasource:

  • You are about to drop the column `name` on the `User` table, which still contains 1 non-null values.

√ Are you sure you want create this migration? ... yes
√ Enter a name for the new migration: ... second
Prisma Migrate created the following migration without applying it 20210917184421_second

You can now edit it and apply it by running prisma migrate dev.
```

And applying it with `migrate deploy` indeed deletes the old columns and creates the new one with a default - without any special warning in the output that some data is going to be deleted (as `deploy` really just executes what is in the migration file).

So probably expected, but not optimal for the user as the message during `--create-only` makes them think something might be deleted but is not, and then later when the data is actually deleted there is no warning.

**Comment by toxaq** [comment]
Just trying to get up to speed with prisma migrations and got this same warning on the '--create-only' flag. Followed Y as per OP and had all my data wiped.

**Comment by janpio** [comment]
That sounds very different from what OP reported @toxaq. They were surprised about a `--create-only` command talking about removing some data. This does not actually happen during the execution of this command - only when you actually deploy the migration afterwards.

Are you sure you got the exact same type of error message on the same command? If so, please open a new issue with a full description of what happened and optimally a reproduction. That would be a bug we would definitely want to investigate.

If `migrate dev --create-only` detected drift though and warned you that it would delete all data, and you confirmed with `y` this would indeed be not super intuitive (We have other issue tracking this) but intended behavior.

**Comment by toxaq** [comment]
Hi @janpio, I ran `npx prisma migrate dev --create-only` and when it detected drift it showed me this message. I spent some time googling to find this thread as I thought it must be wrong. This thread seemed to imply that the message was misleading. It isn't. I pressed Y so all my data was deleted. Coming from Rails this migration system is taking a lot to understand but I suspect that's because I'm trying to make it work like Rails.
